misk/src/main/kotlin/misk/tokens/FakeTokenGenerator.kt

```diff
@@ -1,13 +1,14 @@
 package misk.tokens
 
 import misk.tokens.TokenGenerator.Companion.canonicalize
+import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.atomic.AtomicLong
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class FakeTokenGenerator @Inject constructor() : TokenGenerator {
-  internal val nextByLabel = mutableMapOf<String, AtomicLong>()
+  internal val nextByLabel = ConcurrentHashMap<String, AtomicLong>()
```

Comment:
I claim you almost never want ConcurrentHashMap for things that aren’t extremely highly concurrent. It’s a big slow datastructure designed for very high concurrency. If you only need safety I recommend this:
```
internal val nextByLabel = Collections.synchronizedMap(mutableMapOf<String, AtomicLong>())
```

Reply:
I had considered (googled) both. My understanding is that `ConcurrentHashMap` locks writes and `SynchronizedMap` locks all access, giving the former the behaviour I'm looking for. Is that not the case?

Reply:
CHM is higher concurrency, synchronized map is lower memory. I think for tests the high concurrency won't pay for itself.

Reply:
👍